---
title: "Exp1 BeanSeedSynCom2023"
author: "Matthieu Barret"
date: "2022, March 7th"
output:
  html_document: default
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r directory, echo=FALSE}
setwd("~/Users/garnault-admin/Documents/Thèse/Analyses/SynCom/Metabarcoding")
```

# Data information


# 1 - Sequence analyses 

## 1.1 - Remove primers with cutadapt v1.15

This step is performed outside the R environment
```
for i in `cat group`; do cutadapt --discard-untrimmed -o $i.gyrB.R1.fq -p $i.gyrB.R2.fq -g MGNCCNGSNATGTAYATHGG -G ACNCCRTGNARDCCDCCNGA -e 0.1  -O 20 $i*L001_R1_001.fastq.gz $i*_L001_R2_001.fastq.gz; done
```

## 1.2 - Convert fastq to ASV table (dada2)

To save computing time, the code is not evaluated in the version of the document and the results are not included.

```{r echo=FALSE, eval = FALSE}
library(dada2); packageVersion("dada2")
#Adapt from https://benjjneb.github.io/dada2/tutorial.html
path <- "~/Documents/tmp/data/" 
list.files(path)
fnFs <- sort(list.files(path, pattern="gyrB.R1.fq", full.names = TRUE))
fnRs <- sort(list.files(path, pattern="gyrB.R2.fq", full.names = TRUE))
sample.names <- sapply(strsplit(basename(fnFs), "_"), `[`, 1)
plotQualityProfile(fnFs[1:8]) #select 200 
plotQualityProfile(fnRs[1:8]) #select 150 
filtFs <- file.path(path, "filtered", paste0(sample.names, "_F_filt.fastq.gz"))
filtRs <- file.path(path, "filtered", paste0(sample.names, "_R_filt.fastq.gz"))
names(filtFs) <- sample.names
names(filtRs) <- sample.names
out <- filterAndTrim(fnFs, filtFs, fnRs, filtRs, truncLen=c(200,150),
                     maxN=0, maxEE=c(1,1), truncQ=5, rm.phix=TRUE,
                     compress=TRUE, multithread=TRUE) # 
head(out)
errF <- learnErrors(filtFs, multithread=TRUE)
errR <- learnErrors(filtRs, multithread=TRUE)
plotErrors(errF, nominalQ=TRUE)
dadaFs <- dada(filtFs, err=errF, multithread=TRUE) #without pooling or pseudo-pooling (no need to detect rare ASV)
dadaRs <- dada(filtRs, err=errR, multithread=TRUE)
dadaFs[[1]]
mergers <- mergePairs(dadaFs, filtFs, dadaRs, filtRs, verbose=TRUE)
seqtab <- makeSequenceTable(mergers)
dim(seqtab)
table(nchar(getSequences(seqtab)))
#Since gyrB is a protein-coding genes only triplets should be conserved (244-247-250-253-256-259-262-265-268)
seqtab244 <- seqtab[,nchar(colnames(seqtab)) %in% 244]
seqtab247 <- seqtab[,nchar(colnames(seqtab)) %in% 247]
seqtab250 <- seqtab[,nchar(colnames(seqtab)) %in% 250]
seqtab253 <- seqtab[,nchar(colnames(seqtab)) %in% 253]
seqtab256 <- seqtab[,nchar(colnames(seqtab)) %in% 256]
seqtab259 <- seqtab[,nchar(colnames(seqtab)) %in% 259]
seqtab262 <- seqtab[,nchar(colnames(seqtab)) %in% 262]
seqtab265 <- seqtab[,nchar(colnames(seqtab)) %in% 265]
seqtab268 <- seqtab[,nchar(colnames(seqtab)) %in% 268]
#Merge all files
seq.final <- cbind(seqtab244, seqtab247, seqtab250, seqtab253, seqtab256, seqtab259, seqtab262, seqtab265, seqtab268) 
dim(seq.final)
sum(seq.final)/sum(seqtab)
#Detect/Remove chimera
seqtab.nochim <- removeBimeraDenovo(seq.final, method="consensus", multithread=TRUE, verbose=TRUE)
dim(seqtab.nochim)
sum(seqtab.nochim)/sum(seq.final)
#Detect stop codons
#Summary
getN <- function(x) sum(getUniques(x))
track <- cbind(out, sapply(dadaFs, getN), sapply(dadaRs, getN), sapply(mergers, getN), rowSums(seqtab.nochim))
colnames(track) <- c("input", "filtered", "denoisedF", "denoisedR", "merged", "nonchim")
rownames(track) <- sample.names
head(track)
#Sort ASV by descending order
asvgyrB <- seqtab.nochim[, order(colSums(-seqtab.nochim))]
saveRDS(asvgyrB, "ASV_gyrB.rds")
#Taxonomic affiliation
taxa.g <- assignTaxonomy(asvgyrB, "~/Documents/DB/train_set_gyrB_v4.fa.gz", multithread=TRUE)
taxa.print.g <- taxa.g # Removing sequence rownames for display only
rownames(taxa.print.g) <- NULL
head(taxa.print.g)
saveRDS(taxa.g, "gyrB_taxo.rds")
```


## 1.3 - Construct phyloseq object

```{r phyloseq, echo=FALSE}
library(phyloseq); packageVersion("phyloseq")
design <- read.csv("design_run20220401.csv", sep = ";",  check.names=FALSE, row.names=1)
asvgyrB <- readRDS("ASV_gyrB.rds")
taxgyrB <- readRDS("gyrB_taxo.rds")
psgyrB_0 <- phyloseq(tax_table(taxgyrB), sample_data(design),
                     otu_table(asvgyrB, taxa_are_rows = FALSE)) # 3504 taxa and 286 samples
```

Rename ASVs

```{r, echo=FALSE}
#Rename ASV
dna.gyrB <- Biostrings::DNAStringSet(taxa_names(psgyrB_0))
names(dna.gyrB) <- taxa_names(psgyrB_0)
psgyrB_1 <- merge_phyloseq(psgyrB_0, dna.gyrB)
taxa_names(psgyrB_1) <- paste0("ASV", seq(ntaxa(psgyrB_1)))
```


## 1.4 - Remove contaminants (decontam)

The decontam package provides methods to identify and visualize contaminating DNA features.

Starting by inspecting library sizes

```{r, echo=FALSE}
library(ggplot2) ; packageVersion("ggplot2")
library(decontam); packageVersion("decontam")
df <- as.data.frame(sample_data(psgyrB_1))
df$LibrarySize <- sample_sums(psgyrB_1)
df <- df[order(df$LibrarySize),]
df$Index <- seq(nrow(df))
lib.gyrB <- ggplot(data=df, aes(x=Index, y=LibrarySize, color=Sample_Type)) + geom_point()
lib.gyrB
```

We used is the “prevalence” method where the prevalence (presence/absence across samples) of each sequence feature in true positive samples is compared to the prevalence in negative controls to identify contaminants. We set the threshold to 0.1

```{r, echo=FALSE, results='hide', eval = FALSE}
sample_data(psgyrB_1)$is.neg <- sample_data(psgyrB_1)$Sample_Type == "Negative"
contamdf.prev.gyrB <- isContaminant(psgyrB_1, method="prevalence", neg="is.neg", threshold=0.1)
table(contamdf.prev.gyrB$contaminant) # 8contaminants detected
head(which(contamdf.prev.gyrB$contaminant)) 
ps.pa.gyrB <- transform_sample_counts(psgyrB_1, function(abund) 1*(abund>0))
ps.pa.neg.gyrB <- prune_samples(sample_data(ps.pa.gyrB)$Sample_Type == "Negative", ps.pa.gyrB)
ps.pa.pos.gyrB <- prune_samples(sample_data(ps.pa.gyrB)$Sample_Type == "Sample", ps.pa.gyrB)
df.pa.gyrB <- data.frame(pa.pos=taxa_sums(ps.pa.pos.gyrB), pa.neg=taxa_sums(ps.pa.neg.gyrB),
                    contaminant=contamdf.prev.gyrB$contaminant)
ggplot(data=df.pa.gyrB, aes(x=pa.neg, y=pa.pos, color=contaminant)) + geom_point() +
  xlab("Prevalence (Negative Controls)") + ylab("Prevalence (True Samples)")
```

Removed contaminants and positive and negative controls

```{r, echo=FALSE}
psgyrB_2 <- prune_taxa(!contamdf.prev.gyrB$contaminant, psgyrB_1)
psgyrB_3 <- subset_samples(psgyrB_2, Sample_Type =="Sample")
psgyrB_3 <- filter_taxa(psgyrB_3, function(x) sum(x) > 0, TRUE) #3410 taxa and 270 samples
```


## 1.5 - Filtering

Filter ASV : keep ASVs >3 counts in at least 3 samples (x% of the dataset ) 

```{r, echo=FALSE, results='hide'}
psgyrB_4 <- filter_taxa(psgyrB_3, function(x) sum(x > 3) > (0.01*length(x)), TRUE) # 206 taxa and 270 samples
```

Removed gyrB ASV not affiliated at the kingdom level and phylum level.
Removed gyrB ASV affiliated as parE (a paralog of gyrB)

```{r, echo=FALSE}
psgyrB_5 <- subset_taxa(psgyrB_4, !is.na(Kingdom) & !Kingdom %in% c("parE") & !is.na(Phylum)) ## 172 taxa and 270 samples
```

## 1.6 - Phylogenetic tree
Perform NJ tree for 16S and gyrB - this take a long time so we upload a phyloseq object with the phylogenetic already incorporated.
```{r, echo=FALSE}
library("DECIPHER"); packageVersion("DECIPHER")
library("phangorn"); packageVersion("phangorn")
seqs.gyrB <- refseq(psgyrB_5)
alignment.gyrB <- AlignTranslation(seqs.gyrB, sense = "+", readingFrame = 2, type ="DNAStringSet") 
phang.align.gyrB <- phyDat(as(alignment.gyrB, "matrix"), type="DNA")
dm.gyrB <- dist.ml(phang.align.gyrB)
treeNJ.gyrB <- NJ(dm.gyrB) # Note, tip order != sequence order
dgyrB <- phyloseq(tax_table(psgyrB_5), sample_data(psgyrB_5),
                 otu_table(psgyrB_5, taxa_are_rows = FALSE), refseq(psgyrB_5), phy_tree(treeNJ.gyrB))
saveRDS(dgyrB, "gyrB_PS.rds") 
```

## 1.7 - Assess sequencing depth

```{r, echo=FALSE, message=FALSE}
library(ggplot2)
library(data.table)
qplot(log10(rowSums(otu_table(dgyrB)))) + xlab("Logged counts-per-sample gyrB")

sdtgyrB <- data.table(as(sample_data(dgyrB), "data.frame"),
                      TotalReads = sample_sums(dgyrB), keep.rownames = TRUE)
```

# 2 - Subset for the project (run was made with other projects)


```{r, echo=FALSE, warning=FALSE, message=FALSE, results="hide"}

dgyrB.Syn <- subset_samples(dgyrB, Project =="SynCom")
dgyrB.Syn <- filter_taxa(dgyrB.Syn , function(x) sum(x) > 0, TRUE) #100 taxa and 178 samples

new_sample_data=read.csv(file = "~/Documents/Thèse/Analyses/SynCom/Metabarcoding/Données_run_1er_avril/New_sample_data.csv",h=T,sep=";",row.names = 1)  #type metadata was split into two column -> Inoculation Yes/No ; Maceration Single_seed/Seed_batch
sample_data(dgyrB.Syn)=new_sample_data
#sample_data(data_SynCom)$Concentration = as.numeric(sample_data(data_SynCom)$Concentration)

##remove ASV 33 -> skin bacteria (contamination)
data_SynCom = subset_taxa(dgyrB.Syn, Species != "Cutibacterium_acnes")

saveRDS(data_SynCom, "PS_exp1_filtered.rds")

```









